### PR TITLE
IDForwarding: Add basic metrics

### PR DESCRIPTION
**Pull request by kalleep**
**What is this feature?**
Add metrics to id token signing.
* tokenSigningCounter - Counter newly signed tokens
* tokenSigningFromCacheCounter - Counter of tokens resolved from cache
* failedTokenSingingCounter - Counter of failed token signings
* tokenSigningDurationHistogram - Histogram of time taken for token signing

**Which issue(s) does this PR fix?**:
Part of: https://github.com/grafana/identity-access-team/issues/310

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/auth/idimpl/metrics.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+			Name:      "idforwarding_token_sining_total",
+			Help:      "Number of token signings",
+		}),
+		tokenSigningFromCacheCounter: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: metricsNamespace,
+			Subsystem: metricsSubSystem,
+			Name:      "idforwarding_token_signing_from_cache_total",
+			Help:      "Number of signed tokens retrieved from cahce",
+		}),
+		failedTokenSingingCounter: prometheus.NewCounter(prometheus.CounterOpts{
```
Comment (gamab) [inline pkg/services/auth/idimpl/metrics.go:26]:
```suggestion
		failedTokenSigningCounter: prometheus.NewCounter(prometheus.CounterOpts{
```

File: pkg/services/auth/idimpl/metrics.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+			Namespace: metricsNamespace,
+			Subsystem: metricsSubSystem,
+			Name:      "idforwarding_failed_token_sining_total",
```
Comment (gamab) [inline pkg/services/auth/idimpl/metrics.go:29]:
```suggestion
			Name:      "idforwarding_failed_token_signing_total",
```

File: pkg/services/auth/idimpl/metrics.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	metricsNamespace = "grafana"
+	metricsSubSystem = "idforwarding"
+)
+
+func newMetircus(reg prometheus.Registerer) *metrics {
+	m := &metrics{
+		tokenSigningCounter: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: metricsNamespace,
+			Subsystem: metricsSubSystem,
+			Name:      "idforwarding_token_sining_total",
```
Comment (gamab) [inline pkg/services/auth/idimpl/metrics.go:17]:
```suggestion
			Name:      "idforwarding_token_signing_total",
```

File: pkg/services/auth/idimpl/metrics.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+		failedTokenSingingCounter: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: metricsNamespace,
+			Subsystem: metricsSubSystem,
+			Name:      "idforwarding_failed_token_sining_total",
+			Help:      "Number of failed token singings",
+		}),
+		tokenSigningDurationHistogram: prometheus.NewHistogram(prometheus.HistogramOpts{
+			Namespace: metricsNamespace,
+			Subsystem: metricsSubSystem,
+			Name:      "idfowrading_token_signing_duration_seconds",
```
Comment (gamab) [inline pkg/services/auth/idimpl/metrics.go:35]:
```suggestion
			Name:      "idforwading_token_signing_duration_seconds",
```

File: pkg/services/auth/idimpl/metrics.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+			Namespace: metricsNamespace,
+			Subsystem: metricsSubSystem,
+			Name:      "idforwarding_token_signing_total",
+			Help:      "Number of token signings",
+		}),
+		tokenSigningFromCacheCounter: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: metricsNamespace,
+			Subsystem: metricsSubSystem,
+			Name:      "idforwarding_token_signing_from_cache_total",
+			Help:      "Number of signed tokens retrieved from cahce",
```
Comment (gamab) [inline pkg/services/auth/idimpl/metrics.go:24]:
```suggestion
			Help:      "Number of signed tokens retrieved from cache",
```

**Review by gamab — APPROVED** [review]
This is awesome that we think about metrics at the very beginning of the project.

**Review by Jguer — APPROVED** [review]

**Pull request merged**